app/models/transaction/category.rb

```diff
@@ -46,6 +46,13 @@ def self.create_default_categories(family)
     self.insert_all(categories)
   end
 
+  def reassign_transactions_then_destroy!(replacement)
+    transaction do
+      transactions.update_all category_id: replacement&.id
```

Comment:
This won't run callbacks or expire caches. But we don't need either right now.

If we wanted to account for the above, we could iterate and update each record individually. Would probably do that in a job, too.

I think it's too early to know what we want here. Probably fine to just keep in mind and leave as-is.